### ISSUE TITLE
Fix `larastan` on level 2

### ISSUE THREAD

**Issue by dmohns** [EnAccess/micropowermanager#411]
**Scope:**

Currently `larastan` on level **2** reports errors which indicate broken code in our code base.

For each of the reported errors please decide, whether the corresponding code 

- should be removed as it's never used
- should be fixed as the application will break if executed

**Definition of done:**

The `larastan` level in `phpstan.neon` is set to **2** and the Github Action step `composer run larastan` passes.

**Comment by alaincodes24** [comment]
@dmohns I checked `phpstan.neon` file and the level is set to 1. Would you recommend setting it to 2 and proceed or fix issues related to level 1.

**Comment by dmohns** [comment]
> [@dmohns](https://github.com/dmohns) I checked `phpstan.neon` file and the level is set to 1. `composer run larastan` did not throw any errors but updating the level to 2 shows some errors. I'm working to address those errors, I wanted to let you that it was set to 1 instead of 2.

Yes, this is correct 👍 To provide some context: Ultimately, our goal is to have Larastan set to level 10. However, because going from 0 to 10 requires massive refactoring, we decided to steadily move through Larastan levels, one at a time. We had previously managed to enable it on level 1, see here: https://github.com/EnAccess/micropowermanager/issues/248

We also, added an "optional" CI step that checks Larastan on the next level, i.e. 2 currently, see https://github.com/EnAccess/micropowermanager/pull/410/files

What I would be expecting here, would be one (or multiple) Pull Requests that fix the issues reported by Larastan level 2. Once all are done, the optional `Larastan (next)` step on the CI should pass without errors. Once this has been achieved, we can dump the Larastan level in `phpstan.neon`.

**Comment by dmohns** [comment]
Reopening, as it was closed by mistake. He are still not fully on level 2 yet 😉

**Comment by alaincodes24** [comment]
@dmohns good catch!
